lib/grape/validations/validator_factory.rb

```diff
@@ -1,17 +1,12 @@
 module Grape
   module Validations
     class ValidatorFactory
-      def initialize(**options)
-        @validator_class = options.delete(:validator_class)
-        @options         = options
-      end
-
-      def create_validator
-        @validator_class.new(@options[:attributes],
-                             @options[:options],
-                             @options[:required],
-                             @options[:params_scope],
-                             @options[:opts])
+      def self.create_validator(**options)
+        options.delete(:validator_class).new(options[:attributes],
```

Comment:
Are we modifying the value of caller's `options` here? We need to `dup` here before doing that.

Reply:
The initializer was doing the [same thing](https://github.com/ruby-grape/grape/pull/1898/files/549e1bd84bd39dd63357d12a90a39d23b900279f#diff-8fe683a7109bdcb8bcea665a775a876cL5).

I don't think there is something to worried here, but I can miss something...

```ruby
@validator_class = options.delete(:validator_class)
```

Reply:
I see. We should change it still because that typically has unexpected side effects (we're modifying the caller's hash table).

Reply:
This should just be `options[:validator_class].new`, why are we `delete`-ing it at all?

Reply:
I don't know `delete` is used... I'll change it just like you suggest it.